pylint/message/message_definition_store.py

```diff
@@ -48,11 +48,7 @@ def register_message(self, message: MessageDefinition) -> None:
         self._messages_definitions[message.msgid] = message
         self._msgs_by_category[message.msgid[0]].append(message.msgid)
 
-    # We disable the message here because MessageDefinitionStore is only
-    # initialized once and due to the size of the class does not run the
-    # risk of creating a large memory leak.
-    # See discussion in: https://github.com/PyCQA/pylint/pull/5673
-    @functools.lru_cache()  # pylint: disable=lru-cache-decorating-method
+    @functools.lru_cache()
```

Comment:
Actually I think we should add a high value here, probably twice the size of the current messages lists. (128 is low)

Reply:
Might want to do that in a separate PR/commit

Reply:
We'll have to reinstate the comment explaining why we'd have no cache issue with a big number, and I think the way it is right now is a direct result of the error when reading the doc of `` @functools.lru_cache()`` (we used no value because we thougth no value = infinity) so I think it's relevant / more convenient to do it in this one. I can open another MR though.